LoopKitUI/Views/LevelMaskView.swift

```diff
@@ -16,14 +16,15 @@ public class LevelMaskView: UIView {
 
     var value: Double = 1.0 {
         didSet {
-            if value < 0 || value > 1.0 {
-                assertionFailure("Level mask value outside of interval [0,1]")
-            }
             animateFill(duration: firstDataUpdate ? 0 : 1.25)
             firstDataUpdate = false
         }
     }
 
+    private var boundedValue: Double {
+        return max(0.0, min(1.0, value))
```

Comment:
I often write this extension to encapsulate this pattern:
```swift
extension Comparable {
    func clamped(to range: ClosedRange<Self>) -> Self {
        if self < range.lowerBound {
            return range.lowerBound
        } else if self > range.upperBound {
            return range.upperBound
        } else {
            return self
        }
    }
}
```
Then we get nice call-site clarity like `value.clamped(to: 0...1.0)`.